src/GraphQL.Conventions/Types/Resolution/ICanChangeReflectorSettings.cs

```diff
@@ -0,0 +1,65 @@
+﻿using GraphQL.Conventions.Types.Resolution;
+using System.Linq;
+
+namespace GraphQL.Conventions
+{
+    public interface ICanChangeReflectorSettings
+    { }
+
+    public static class ReflectorSettingsExtensions
+    {
+        static object _lockKey = new object();
+
+        public static T IgnoreTypesFromNamespacesStartingWith<T>(this T source, params string[] namespacesToIgnore) where T : ICanChangeReflectorSettings
+        {
+            if (namespacesToIgnore == null || !namespacesToIgnore.Any())
+                return source;
+
+            namespacesToIgnore = namespacesToIgnore.Distinct().ToArray();
+            foreach (var @namespace in namespacesToIgnore)
+                if (!ObjectReflectorSettings.IgnoredNamespaces.Contains(@namespace))
```

Comment:
This bit isn't thread-safe as there could potentially be an entry inserted before you reach the lock on line 22 below.